hal/inc/concurrent_hal.h

```diff
@@ -128,6 +131,25 @@ os_result_t os_thread_cleanup(os_thread_t thread);
  */
 os_result_t os_thread_yield(void);
 
+os_unique_id_t os_thread_unique_id(os_thread_t thread);
+
+os_thread_t os_thread_current();
+
+typedef struct {
+  uint8_t reserved;
+
+  os_thread_t thread;
+  const char* name;
+  os_unique_id_t id;
+  void* stack;
+  void* stack_start;
+  void* stack_end;
+} os_thread_dump_info_t;
```

Comment:
If we wanted to expand this structure later while retaining backwards compatibility, what would we need to add/change now?  A reserved void * at the end?

Reply:
It's currently only used within a single module and is not exported, so that shouldn't be a concern at the moment. If we later decide to export it, the first `reserved` becomes a `version` field as usual.

Reply:
Thanks for pointing that out @avtolstoy, sounds good.